### PR TITLE
ElectionList over PdfMake

### PR DESCRIPTION
**Pull request by tsiegleauq**
Creates the election list over PdfMake. There is one line that hopefully does not cause too much headache

### CODE REVIEW

File: openslides/assignments/static/js/assignments/site.js [hunk 1]
```diff
@@ -40,6 +40,9 @@ angular.module('OpenSlidesApp.assignments.site', [
                     assignments: function(Assignment) {
                         return Assignment.findAll();
                     },
+                    users: function(User) {     //TODO: Load only related users cause of better performance
+                        return User.findAll();
+                    },
```
Comment (tsiegleauq) [inline openslides/assignments/static/js/assignments/site.js]: This is _a_ solution for [this isse](https://github.com/OpenSlides/OpenSlides/issues/2581).
What would _probably_ be a better solution would be something like

```
angular.forEach($scope.assignmentsFiltered, function(assignment) {
    Assignment.loadRelations(assignment, 'assignment_related_users');
    ...
});
```
after the click on the PDF-Button. That should only re-load users needed for the elections. Unfortionatly, this causes an error.

Comment (normanjaeckel) [inline openslides/assignments/static/js/assignments/site.js]: Which error?

Comment (tsiegleauq) [inline openslides/assignments/static/js/assignments/site.js]: `openslides-libs.js:5 Error: assignments/relateduser.inject: "attrs" must be an object or an array!
Error`

**Comment by emanuelschuetze** [comment]
I rebased on top of master and made some (code) styling improvements on motions and assignments pdfs.

But I does not find a solution to replace `User.findAll()` in resolve of AssignmentListCtrl. I'm not sure if it is possible with the Assignment.loadRelations(..) method [http://www.js-data.io/docs/dsloadrelations].
@normanjaeckel Do you have any idea how to load relations for `assignment_related_users` (which is a relation of a relation)? (assignment -> assignment_related_users -> user)

Maybe we should split this problem out from this pull request...

File: openslides/assignments/static/js/assignments/site.js (hunk 1 above)
Comment (normanjaeckel) [inline openslides/assignments/static/js/assignments/site.js]: Try `Assignment.loadRelations(assignment,  ['assignment_related_users'])`

File: openslides/assignments/static/js/assignments/site.js [hunk 2]
```diff
@@ -38,11 +38,11 @@ angular.module('OpenSlidesApp.assignments.site', [
...
                 resolve: {
                     assignments: function(Assignment) {
-                        return Assignment.findAll();
-                    },
-                    users: function(User) {
-                        //TODO: Load only related users cause of better performance
-                        return User.findAll();
+                        return Assignment.findAll().then(function(assignments) {
+                            angular.forEach(assignments, function(assignment) {
+                                Assignment.loadRelations(assignment, ['assignment_related_users']);
```
Comment (emanuelschuetze) [inline openslides/assignments/static/js/assignments/site.js]: @normanjaeckel Same error if I use this line to load all related users. Any idea?

**Comment by normanjaeckel** [comment]
You can try to use loadRelations without any field argument, see [here](http://www.js-data.io/docs/dsloadrelations). If this does not work, I propose the following workaround: Load User.findAll() not in the state resolver but inside `$scope.makePDF_assignmentList`. Then create the PDF inside the `then()` function. We we do not load all users from beginning but only if someone clicks on make PDF button. Also an issue to correct this workaround is necessary.

**Comment by emanuelschuetze** [comment]
> You can try to use loadRelations without any field argument

Unfortunately, this does not work:
```
Error: assignments/relateduser.inject: "attrs" must be an object or an array!
Error: assignments/poll.inject: "attrs" must be an object or an array!
```

I implemented the workaround: Load all Users in pdfmake function - not in state resolver of list view.

@normanjaeckel Please review and merge.

**Comment by emanuelschuetze** [comment]
> I implemented the workaround: Load all Users in pdfmake function - not in state resolver of list view.

see https://github.com/OpenSlides/OpenSlides/pull/2610/files#diff-743ecf6da36e2c2e458d718572e809e1R349

**Comment by emanuelschuetze** [comment]
@normanjaeckel Ping, I need this pull request in master to use it for my next pdf improvements.

File: openslides/assignments/static/js/assignments/site.js [hunk 3]
```diff
@@ -338,6 +344,25 @@ angular.module('OpenSlidesApp.assignments.site', [
...
+        // create the PDF List
+        $scope.makePDF_assignmentList = function () {
+            User.findAll().then( function(users) {
+                var filename = gettextCatalog.getString("Elections") + ".pdf";
+                var assignmentContentProviderArray = [];
+
+                //convert the filtered assignments to content providers
+                angular.forEach($scope.assignmentsFiltered, function(assignment) {
+                    assignmentContentProviderArray.push(
+                            AssignmentContentProvider.createInstance(assignment));
```
Comment (normanjaeckel) [inline openslides/assignments/static/js/assignments/site.js]: Indention is broken.

Comment (emanuelschuetze) [inline openslides/assignments/static/js/assignments/site.js]: done

**Comment by emanuelschuetze** [comment]
Also fixed two travis errors in core/config.py and utils/search.py because travis used the latest flake8 release.

**Pull request merged**